File: IntegrationGuideForAndroid/Apps/brandStaticDelegateJava/src/main/java/com/example/integrationguide/MainActivity.java
// Copyright 2023 VMware, Inc.
// SPDX-License-Identifier: BSD-2-Clause

package com.example.integrationguide;

import android.Manifest;
import android.app.Activity;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import com.airwatch.sdk.context.SDKContext;
import com.airwatch.sdk.context.SDKContextManager;

import java.util.HashMap;

public class MainActivity extends BaseActivity {

    private static final int NOTIFICATION_REQ_CODE = 101;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        configureTextView();

        toastHere(configureStatus());
        setUpPermissions();
    }

    private void setUpPermissions() {
        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            int permission = ContextCompat.checkSelfPermission(
                    this,
                    Manifest.permission.POST_NOTIFICATIONS
            );

            if (permission != PackageManager.PERMISSION_GRANTED) {
                ActivityCompat.requestPermissions(
                        this,
                        new String[]{Manifest.permission.POST_NOTIFICATIONS},
                        NOTIFICATION_REQ_CODE
                );
            }
        }
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        if (requestCode == NOTIFICATION_REQ_CODE){
            if (grantResults.length == 0 || grantResults[0] != PackageManager.PERMISSION_GRANTED) {
                toastHere("Notification Permission has been denied by user");
            } else {
                toastHere("Notification Permission has been granted by user");
            }
        }
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
    }

    private void toastHere(final String message) {
        final Activity activity = this;
        runOnUiThread(new Runnable() {
            @Override
            public void run() {
                Toast.makeText(activity, message, Toast.LENGTH_LONG).show();
            }
        });
    }

    public String configureStatus() {
        final SDKContext sdkContext = SDKContextManager.getSDKContext();
        final String status = getString(
            R.string.sdk_state, sdkContext.getCurrentState().name());

        final TextView textView = (TextView) findViewById(
            R.id.textViewConfiguration);
        textView.setText(status);
        textView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) { statusToggle(); }
        });

        final HashMap<String, String> hashMap = new HashMap<String, String>();
        hashMap.put(
            "c", Integer.toString(sdkContext.getCurrentState().getValue()));
        hashMap.put("n", sdkContext.getCurrentState().name());
        hashMap.put(
            "I", Integer.toString(SDKContext.State.INITIALIZED.getValue()));
        hashMap.put(
            "C", Integer.toString(SDKContext.State.CONFIGURED.getValue()));

        final String[] message = new String[] {
           sdkContext.getCurrentState() == SDKContext.State.IDLE ?
               getString(R.string.sdk_configuration_unavailable) :
               sdkContext.getSDKConfiguration().toString(),
            hashMap.toString()
        };
        ((TextView) findViewById(R.id.textViewScrolling))
            .setText(TextUtils.join("\n", message));

        return status;
    }

    private void statusToggle() {
        final View view = findViewById(R.id.scrollView);
        final boolean visible = (view.getVisibility() == View.VISIBLE);
        view.setVisibility(visible ? View.GONE : View.VISIBLE);
        findViewById(R.id.toggleView).setVisibility(
            visible ? View.VISIBLE : View.GONE);
    }

}

File: IntegrationGuideForAndroid/Apps/brandDynamicDelegateKotlin/src/main/java/com/example/integrationguide/MainActivity.kt
// Copyright 2023 VMware, Inc.
// SPDX-License-Identifier: BSD-2-Clause

package com.example.integrationguide

import android.Manifest
import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.core.app.ActivityCompat
import androidx.core.app.NotificationCompat
import androidx.core.app.NotificationManagerCompat
import androidx.core.content.ContextCompat
import com.airwatch.sdk.SDKManager
import com.airwatch.sdk.context.SDKContext
import com.airwatch.sdk.context.SDKContextManager

class MainActivity : BaseActivity() {
    companion object {
        val channelID = "CHANNEL"
        private const val NOTIFICATION_REQ_CODE = 101
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        configureTextView()
        setUpPermissions()
        if (
            Build.VERSION.SDK_INT >= Build.VERSION_CODES.O
        ) {
            val channel = NotificationChannel(
                channelID, "Channel", NotificationManager.IMPORTANCE_DEFAULT
            )
                .also { it.description = "Diagnostic channel" }
            val manager = applicationContext.getSystemService(
                Context.NOTIFICATION_SERVICE
            ) as NotificationManager
            manager.createNotificationChannel(channel)

            findViewById<View>(R.id.toggleView).setOnClickListener {
                val notification = NotificationCompat.Builder(this, channelID)
                    .setSmallIcon(R.drawable.brand_logo_onecolour)
                    .setContentTitle("Tapped")
                    .setContentText("App Logo was tapped.")
                    .setPriority(NotificationCompat.PRIORITY_DEFAULT)
                with(NotificationManagerCompat.from(this)) {
                    notify(1, notification.build())
                }
            }
        }

        toastHere(configureStatus())

        val enterpriseLogoImageView = findViewById<ImageView>(R.id.imageViewEnterpriseLogo)
        enterpriseLogoImageView.post {
            BrandingManager.getInstance(this).defaultBrandingManager.brandLoadingScreenLogo({ bitmap ->
                enterpriseLogoImageView.setImageBitmap(bitmap)
            }, enterpriseLogoImageView.width, enterpriseLogoImageView.height)
        }

    }

    private fun setUpPermissions() {
        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            val permission = ContextCompat.checkSelfPermission(
                this,
                Manifest.permission.POST_NOTIFICATIONS
            )

            if (permission != PackageManager.PERMISSION_GRANTED) {
                ActivityCompat.requestPermissions(
                    this,
                    arrayOf(Manifest.permission.POST_NOTIFICATIONS),
                    NOTIFICATION_REQ_CODE
                )
            }
        }
    }

    override fun onRequestPermissionsResult(requestCode: Int,
                                            permissions: Array<String>, grantResults: IntArray) {
        when (requestCode) {
            NOTIFICATION_REQ_CODE -> {

                if (grantResults.isEmpty() || grantResults[0] != PackageManager.PERMISSION_GRANTED) {
                    toastHere("Notification Permission has been denied by user")
                } else {
                    toastHere("Notification Permission has been granted by user")
                }
            }
        }
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }

    private fun configureStatus():String {
        val sdkContext = SDKContextManager.getSDKContext()
        val sdkManager = SDKManager.init(applicationContext)
        val status = getString(
            R.string.sdk_state, sdkContext.currentState.name)

        findViewById<TextView>(R.id.textViewConfiguration).let {
            it.text = status
            it.setOnClickListener { statusToggle() }
        }

        val message = listOf(
            if (sdkContext.currentState == SDKContext.State.IDLE)
                getString(R.string.sdk_configuration_unavailable)
            else
                sdkContext.sdkConfiguration.toString(),
            mapOf(
                "c" to sdkContext.currentState.value,
                "n" to sdkContext.currentState.name,
                "I" to SDKContext.State.INITIALIZED.value,
                "C" to SDKContext.State.CONFIGURED.value
            ).toString()
        ).joinToString(separator = "\n")

        // Dump the message into the textView that is inside the scrolling
        // portion.
        findViewById<TextView>(R.id.textViewScrolling).text = message

        return status
    }

    fun statusToggle() {
        val visible = findViewById<View>(R.id.scrollView).let { scrollView ->
            (scrollView.visibility == View.VISIBLE).also {
                scrollView.visibility = if (it) View.GONE else View.VISIBLE
            }
        }
        findViewById<View>(R.id.toggleView).visibility =
            if (visible) View.VISIBLE else View.GONE
    }

    private fun toastHere(message: String) { runOnUiThread {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show() }}
}
